calculateSNR.m

```diff
@@ -286,86 +305,88 @@
+        % will cut segments around the target frequency and each harmonic from
+        % the long spectra. We will average these segments and from this
+        % avergae, we will extract the zscore (signal vs. noise). 
+        
+        % cut out segments of frequency bins around each frequency of interest
+        % (target and its harmonics) and average these segments 
+        mXavgHarmonics = getAvgHarmonics(mX, cfg.idxHarmonics, cfg.binSize);
 
+        % define indices of positions in the averaged segment that represent
+        % noise (note that the center position is the frequency bin where we
+        % expect signal, and noise is on either side of this bin) 
         noiseFs = [(cfg.binSize + 1 - cfg.binSize / 2 - cfg.gap) ...
                    :(cfg.binSize + 1 - 1 - cfg.gap) ...
                    (cfg.binSize + 1 + 1 + cfg.gap) ...
                    :(cfg.binSize + 1 + cfg.binSize / 2 + cfg.gap)];
 
+        % extract magnitudes of signal and noise from the average segment 
         AmpNoise = mXavgHarmonics(noiseFs, :);
         NoiseMean = mean(AmpNoise, 1);
         NoiseSD = std(AmpNoise, 0, 1);
+        % get zscore 
         targetHarmonicsZ = (mXavgHarmonics(cfg.binSize + 1, :) - NoiseMean) ./ NoiseSD;
 
         % save map as nii
         newFileName = ['AvgZHarmonics_', boldFileName, '.nii'];
-        writeMap(targetHarmonicsZ, maskHdr, maskImg, newFileName, destinationDir);
-        %     writeMap(targetHarmonicsZ, maskHdr.fname, newFileName, destinationDir);
-
-        % plot best voxels
-        [~, idxSorted] = sort(targetHarmonicsZ, 'descend');
-        idxSorted(isnan(targetHarmonicsZ(idxSorted))) = [];
-        nMax = 10;
-
-        f = figure('color', 'white', 'Position', [131 728 1744 140]);
-        pnl = panel(f);
-        pnl.pack('h', nMax);
-
-        for iVox = 1:nMax
-            mXbest = mXavgHarmonics(:, idxSorted(iVox));
-            pnl(iVox).select();
-            h = stem(mXbest, ...
-                     'marker', 'none', ...
-                     'color', [0.6, 0.6, 0.6], ...
-                     'linew', 4);
-            hold on;
-            h = stem(cfg.binSize + 1, mXbest(cfg.binSize + 1), ...
-                     'marker', 'none', ...
-                     'color', 'r', ...
-                     'linew', 4);
-            title(sprintf('z=%.2f  vox=%d', targetHarmonicsZ(idxSorted(iVox)), ...
-                          idxSorted(iVox)));
-            set(gca, 'xtick', []);
-        end
-
-        pnl.ylabel('amplitude');
-        pnl.marginbottom = 3;
+        zHarmonicsImg = writeMap(targetHarmonicsZ, maskHdr, maskImg, newFileName, destinationDir);
 
+        coordHarmonics = getVoxelCoordinate(boldHdr, zHarmonicsImg, maskImg, voxelNbToPlot);
+        
+        % plot averge segment from the spectra for best voxels
+        % for a clear representation, does it make sense to use fixed scale
+        % across voxels?
```

Comment:
this is my question: if y-axis scaling is fixed, would it be more visually striking to see which voxels are high and then it is followed by lower z-score values? Now all the y axis is scaled according to the input. and they all look high, but actually couple of them providing low y-value. Does it make sense?